tracks/typescript/exercises/rna-transcription/mentoring.md

```diff
@@ -56,3 +56,29 @@ sequence.replace(/[CGAT]g/, (nucleotide) => /* */)
 
 - Encourage streams because it's harder to use intermediary variables (which can lead to bugs) and forces "one-by-one" approach. It is easier to explain the [`Object`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) solution over multiple conditionals when it's a stream.
 - If a student uses an `Object`, point them to `falsy` values.
+- Highlight TypeScript's ability to infer types automatically, reducing the need for explicit type declarations in many cases.
+  - For example, TypeScript can infer the types of objects like TRANSCRIPTION without the need for explicit type declarations, making code more concise and readable. For instance, the following code:
+    ```
+    const TRANSCRIPTION = {
+      C: 'G',
+      G: 'C',
+      A: 'U',
+      T: 'A',
+    };
+    ``` 
+    is functionally equivalent to the more verbose version with explicit type declarations:
```

Comment:
Whilst I appreciate the addition, and I think inferring types is good (I use it all the time myself), the two are [_not_ functionally equivalent](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAKgJQIIDkDKBhBBJACnbAeRRgF4YBvAKBhgwC4YByAcSYBoaYXGmMOuSXgFUBtAPTj4vJBxiSYSADZKQAdwgwAZiABOAcwCmUKAEsw+mADcAhkoCuhiIJXqIvOCEUATbzGFgANZg6mDWdo4QTFQAvgDcVFRQAJ4ADoYwACJgNij2wEqGIGbeGeRMsjAAPsz81cxs9Uxw0SnpMAg5eQVFJWXMlTV8TE2NQ6KJoJCwcLo2kMC6pqlm4ACyphAQ5vqEYIaMCIagut4APNm5+YXFpqXsHV3XvXeGAHxklFwMDWLcvPxOLQhMxRJwYpNwNB4HMFksVqZwEhfDs9gcOsc9OdLt0bn0Hp0rj1bqUPuRqLQfqw-jxan8QUwwVw4DI-pkQE4UMVlKo1B4vMi-PYgiE1GFbA4nNEIVQptDZvMIItlqswHA0iBDpjThcnsT8Y8iXjXmSvpTeGwgf86VaGUzaCzmAANaWQ6YwxXKhHgABS9mgCFM+gAFlAtSdsXrjfdDbiXqTPhS6BaaQD6SI-o6KgIYkA).

Whilst a more verbose solution here may not make sense, if `DnaNucleotide` and `RnaNucleotide` are externally provisioned, it could make a lot of sense to define the type as `Record<X, Y>`. But if those types exist, one may also expect all the transitions/transcriptions to exist as type.